ast/src/main/java/com/graphicsfuzz/common/tool/PrettyPrinterVisitor.java

```diff
@@ -460,23 +464,22 @@ public void visitWhileStmt(WhileStmt whileStmt) {
 
   @Override
   public void visitForStmt(ForStmt forStmt) {
-    out.append(indent() + "for(" + newLine());
-    out.append("    ");
+    insideForStatement = true;
+    out.append(indent() + "for(");
     visit(forStmt.getInit());
-    out.append("    " + indent());
+    out.append(" ");
     if (forStmt.hasCondition()) {
       visit(forStmt.getCondition());
     }
-    out.append(";" + newLine());
-    out.append("    " + indent());
+    out.append("; ");
     if (forStmt.hasIncrement()) {
       visit(forStmt.getIncrement());
     }
-    out.append(newLine());
-    out.append(indent() + ")" + newLine());
+    out.append(" )" + newLine());
     increaseIndent();
     visit(forStmt.getBody());
     decreaseIndent();
+    insideForStatement = false;
```

Comment:
I think you probably want this `insideForStatement = false;` to go higher up, otherwise declaration statements inside the for loop body won't get a new line.

```
for(;;;){
int i;int y;int z;
}
```

That is my guess anyway. 

The boolean would also get overwritten to false after a nested for loop, so I assume this would look OK:

```
for(;;;){
for(;;;){}
int i;
int y;
int z;
}
```

Maybe rename it `insideForInitStatement` and just set it to true during `visit(forStmt.getInit());`.

Reply:
Done.